databricks/koalas/series.py

```diff
@@ -3409,6 +3409,90 @@ def replace(self, to_replace=None, value=None, regex=False) -> 'Series':
 
         return self._with_new_scol(current)
 
+    def update(self, other):
+        """
+        Modify Series in place using non-NA values from passed Series. Aligns on index.
+
+        Parameters
+        ----------
+        other : Series
+
+        Examples
+        --------
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, 5, 6]))
+        >>> s
+        0    4
+        1    5
+        2    6
+        Name: 0, dtype: int64
+
+        >>> s = ks.Series(['a', 'b', 'c'])
+        >>> s.update(ks.Series(['d', 'e'], index=[0, 2]))
+        >>> s
+        0    d
+        1    b
+        2    e
+        Name: 0, dtype: object
+
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, 5, 6, 7, 8]))
+        >>> s
+        0    4
+        1    5
+        2    6
+        Name: 0, dtype: int64
+
+        >>> s = ks.Series([1, 2, 3], index=[10, 11, 12])
+        >>> s
+        10    1
+        11    2
+        12    3
+        Name: 0, dtype: int64
+
+        >>> s.update(ks.Series([4, 5, 6]))
+        >>> s
+        10    1
+        11    2
+        12    3
+        Name: 0, dtype: int64
+
+        >>> s.update(ks.Series([4, 5, 6], index=[11, 12, 13]))
+        >>> s
+        10    1
+        11    4
+        12    5
+        Name: 0, dtype: int64
+
+        If ``other`` contains NaNs the corresponding values are not updated
+        in the original Series.
+
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, np.nan, 6]))
+        >>> s
+        0    4.0
+        1    2.0
+        2    6.0
+        Name: 0, dtype: float64
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, Series):
+            raise ValueError("'other' must be a Series")
+
+        index_scol_name = self._index_map[0][0]
+        combined = combine_frames(self._kdf, other._kdf, how='leftouter')
+        combined_sdf = combined._sdf.sort(index_scol_name)
```

Comment:
Do we need to sort here?

Reply:
yeah, because the index order is not guaranteed after `combine_frames`

```python
>>> s1 = ks.Series([1, 2, 3], index=[10, 11, 12])
>>> s2 = ks.Series([4, 5, 6], index=[11, 12, 13])
>>> combined = combine_frames(s1.to_frame(), s2.to_frame(), how='leftouter')
>>> combined
   this that
      0    0
10    1  NaN
12    5  5.0
11    4  4.0
```

Reply:
We don't guarantee the row order, so I don't think we need to sort here.

Reply:
oh i got it, then below example is okay??

```python
>>> s = ks.Series([1, 2, 3], index=[10, 11, 12])
>>> s
10    1
11    2
12    3
Name: 0, dtype: int64

>>> s.update(ks.Series([4, 5, 6]))
>>> s
10    1
12    3
11    2
Name: 0, dtype: int64
```